eng/validation/update-channel.ps1

```diff
@@ -14,56 +15,72 @@ function Get-Headers([string]$accept, [string]$barToken) {
-# was already validated and added to $targetChannelName. So we need to check if the build is already in $targetChannelName so we don't fail trying to
-# update the channel each time
-$getBuildApiEndpoint = "$maestroEndpoint/api/builds/$buildId/?api-version=$apiVersion"
-$build = Invoke-WebRequest -Uri $getBuildApiEndpoint -Headers $headers | ConvertFrom-Json
-$buildInChannel = $($build.channels | Where-Object -Property "name" -Value "${targetChannelName}" -EQ)
+    . $PSScriptRoot\..\common\darc-init.ps1
 
-if ($buildInChannel) {
-    Write-Host "Build '$buildId' is already in channel '$targetChannelName'. This is most likely an arcade-validation internal build"
-    exit 0
-}
+    $DarcOutput = darc add-build-to-channel --id $buildId --channel "$targetChannelName"
+    
+    if ($LastExitCode -ne 0) {
+        Write-Host "Problems using Darc to promote build ${buildId} to channel ${targetChannelName}. Stopping execution..."
+        Write-Host $DarcOutput
+        exit 1
+	}
 
-$getChannelsEndpoint = "$maestroEndpoint/api/channels?api-version=2018-07-16"
-$channels = Invoke-WebRequest -Uri $getChannelsEndpoint -Headers $headers | ConvertFrom-Json
-$matchedChannel = $($channels | Where-Object -Property "name" -Value "${targetChannelName}" -EQ | Select-Object -Property id)
+    if ($DarcOutput -match "has already been assigned to") 
+    {
+        Write-Host "Build has already been assigned to channel '${targetChannelName}'."
+	}
+    else {
+	    $buildUrlRegex = "https://dnceng.visualstudio.com/internal/_build/results\?buildId=(?<buildId>[0-9]*)"
 
-if (!$matchedChannel) {
-    Write-Host "Channel with name '$targetChannelName' was not found..."
-    exit 1    
-}
+	    $azdoBuildId = $DarcOutput | select-string -Pattern $buildUrlRegex -AllMatches | % { $_.Matches.Groups[1].Value } 
+        $waitIntervalsInSeconds = 60
+        $build = $null
 
-$channelId = $matchedChannel.id
+	    do {
+            Write-Host "Waiting ${waitIntervalsInSeconds} seconds for promotion build to complete... https://dnceng.visualstudio.com/internal/_build/results?buildId=${azdoBuildId}"
 
-try {
-    $postBuildIntoChannelApiEndpoint = "$maestroEndpoint/api/channels/$channelId/builds/$buildId/?name=$arcadeSdkPackageName&version=$arcadeSdkVersion&api-version=2018-07-16"
-    $headers = Get-Headers 'application/json' $barToken
-        
-    Write-Host "POSTing to $postBuildIntoChannelApiEndpoint..."
-    $postResponse = Invoke-WebRequest -Uri $postBuildIntoChannelApiEndpoint -Headers $headers -Method Post
-    Write-Host "Build '$buildId' was successfully added to channel '$channelId'"
+            Start-Sleep -Seconds $waitIntervalsInSeconds
+
+            $build = Get-AzDO-Build -token $azdoToken -azdoBuildId $azdoBuildId
+	    } while ($build.status -ne "completed")
+
+        if ($build.result -eq "succeeded") {
+            Write-Host "The build was promoted with success."
```

Comment:
I liked the old message better: `"Build '$buildId' was successfully added to channel '$channelId'"`